### PR TITLE
Utils and Basic UI

### PR DESCRIPTION
**Pull request by ArneSchulze**
Hey guys,

please review my code. Thanks

### CODE REVIEW

File: Client/app/src/main/java/de/ameyering/wgplaner/wgplaner/structure/Item.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    public String name;
+    public User requestedBy;
+    public User requestedFor;
+    private float price;
+    private String bougthOn;
+
+    public Item(String name, User requestedBy, User requestedFor) {
+        this.name = name;
+        this.requestedBy = requestedBy;
+        this.requestedFor = requestedFor;
```
Comment (archer96) [inline Client/app/src/main/java/de/ameyering/wgplaner/wgplaner/structure/Item.java]: Better use the other overload like `Item(name, requestedBy, requestedFor, 0f, "")`. This way we avoid duplicate code.

File: Client/app/src/main/java/de/ameyering/wgplaner/wgplaner/structure/Item.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * Created by D067867 on 06.10.2017.
+ */
+
+public class Item {
+    public String name;
+    public User requestedBy;
+    public User requestedFor;
+    private float price;
```
Comment (archer96) [inline Client/app/src/main/java/de/ameyering/wgplaner/wgplaner/structure/Item.java]: We should not use `float` for prices. Maybe `long` for cents or maybe introduce another type `Money`?

*See: http://www.javapractices.com/topic/TopicAction.do?Id=213*

File: Client/app/src/main/java/de/ameyering/wgplaner/wgplaner/structure/Item.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        this.bougthOn = bougthOn;
+    }
+
+    public boolean buy(float price){
+        if(price > 0){
+            this.price = price;
+            this.bougthOn = DataContainer.Item.getDateFormat().format(Calendar.getInstance().getTime()); //Formats actual Time to String
+            return true;
+        }
+        else{
```
Comment (archer96) [inline Client/app/src/main/java/de/ameyering/wgplaner/wgplaner/structure/Item.java]: Remove the `else`. `return false` is enough.

File: Client/app/src/main/java/de/ameyering/wgplaner/wgplaner/structure/User.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                return true;
+            }
+            else{
+                return false;
+            }
+        }
+        else{
+            return false;
+        }
+    }
```
Comment (archer96) [inline Client/app/src/main/java/de/ameyering/wgplaner/wgplaner/structure/User.java]: You could replace all of those `if`s with 

```java
return obj.getClass() == this.getClass() && this.uid.equals(((User) obj).uid);
```

File: Client/app/src/main/java/de/ameyering/wgplaner/wgplaner/utils/Configuration.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+import java.io.Serializable;
+import java.util.HashMap;
+
+/**
+ * Created by D067867 on 06.10.2017.
+ */
+
+public class Configuration implements Serializable{
+    public static Configuration singleton;
+    public static final long serialVersionUID = 7526471155622776137L; //Needed for Serialization (ObjectOutputStream)
```
Comment (archer96) [inline Client/app/src/main/java/de/ameyering/wgplaner/wgplaner/utils/Configuration.java]: `7526471155622776137L` looks like a magic number. What does this number represent?

File: Client/app/src/main/java/de/ameyering/wgplaner/wgplaner/utils/Configuration.java [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+        //Declare Config-Tye here
+        NAME;
+    }
+
+    /*
+    FileHandler-class for Configuration
+    Uses ObjectInputStream and ObjectOutputStream to convert read data easily into objects
+     */
+
+    private static class ConifgFileHandler{
```
Comment (archer96) [inline Client/app/src/main/java/de/ameyering/wgplaner/wgplaner/utils/Configuration.java]: Typo `Conifg != Config`
Should be `ConfigFileHandler`

File: Client/app/src/main/java/de/ameyering/wgplaner/wgplaner/section/splashScreen/SplashScreen.java [hunk 7]
```diff
@@ -20,13 +21,21 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
...
+                Intent intent = new Intent(SplashScreen.this, Home.class);
+                startActivity(intent);
+                finish();
+            }
+        }).start();
 
-        Intent intent = new Intent(this, LoginActivity.class);
-        startActivity(intent);
-        finish();
+        getData();
```
Comment (archer96) [inline Client/app/src/main/java/de/ameyering/wgplaner/wgplaner/section/splashScreen/SplashScreen.java]: Remove duplicate.

File: Client/app/src/main/java/de/ameyering/wgplaner/wgplaner/structure/User.java (hunk 4 above)
Comment (ArneSchulze) [reply Client/app/src/main/java/de/ameyering/wgplaner/wgplaner/structure/User.java]: Isn't it difficult to read?

Comment (archer96) [reply Client/app/src/main/java/de/ameyering/wgplaner/wgplaner/structure/User.java]: I think it's better to read than those nested ifs.

**Comment by archer96** [comment]
Great, thanks!
But please remove the `created by` comments and add some error handling (empty catch clause is insufficient). :smile:

**Pull request merged**